lib/filter-stream.js

```diff
@@ -122,13 +127,17 @@ FilterStream.HIGH_WATER_MARK = 64 * 1024
 
     this._hash = crypto.createHash( this.blockMap.checksumType )
 
-    if( range.checksum !== digest ) {
+    if( this.options.verify && (range.checksum !== digest) ) {
       error = new Error( `Invalid checksum for range [${range.startLBA},${range.endLBA}], bytes ${range.start}-${range.end}` )
       error.checksum = digest
       error.range = range
       this.emit( 'error', error )
     }
 
+    if ( this.options.calculateChecksums ) {
+      range.range.checksum = digest
```

Comment:
Hmm? How come `this.options.verify` checks for `(range.checksum !== digest)`, but `this.options.calculateChecksums` sets `range.range.checksum = digest` ?  (apologies if that's a stupid question)

Reply:
`ReadRange`s are created from `Range`s and their checksum is copied.
See https://github.com/resin-io-modules/blockmap/blob/master/lib/read-range.js#L20 .